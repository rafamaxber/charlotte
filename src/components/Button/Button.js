import React from 'react';
import styled from 'styled-components';

const colorsForButton = {
  default: "#F98100",
  info: "#79BD1A",
}

const Button = styled.a`
  font-family: 'Heebo', sans-serif;
  font-style: normal;
  font-weight: bold;
  font-size: 15px;
  max-width: 200px;
  background-color: #FFF;
  border-width: 2px;
  border-style: solid;
  border-color: ${props => colorsForButton[props.typeBtn || 'default']};
  border-radius: 25px;
  padding: 12px 43px;
  text-align: center;
  display: inline-block;
  line-height: normal;
  text-align: center;
  text-decoration: none;
  outline: none;
  cursor: pointer;
  color: ${props => colorsForButton[props.typeBtn || 'default']};
  transition: .3s ease;
  &:hover, &:focus {
    background-color:${props => colorsForButton[props.typeBtn || 'default']};
    color: #fff;
  }
`;

const ButtonSmall = Button.extend`
  font-size: 13px;
  border-width: 1px;
  padding: 10px 24.5px;
`;

export default ({ typeBtn, text, children, size, action, href }) => {
  if (size === 'small') {
    return (
      <ButtonSmall typeBtn={typeBtn} onClick={action}>
        {children}
      </ButtonSmall>
    );
  }
  return (
    <Button typeBtn={typeBtn}>
      {children}
    </Button>
  );
};
