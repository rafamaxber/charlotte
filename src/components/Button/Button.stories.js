import React from 'react';
import { storiesOf } from '@storybook/react';
import Button from './Button';

storiesOf('Button', module)
  .add('Button default shaped', () => (
    <Button typeBtn='default'></Button>
  ))
  .add('Button default green', () => (
    <Button typeBtn='level2' text='Book now'></Button>
  ))
  .add('Button default shaped small', () => (
    <Button typeBtn='default' btnSize='small' text='Price history'></Button>
  ));
