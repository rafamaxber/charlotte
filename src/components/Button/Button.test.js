import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('Button', () => {
  it('Should match with snapshot', () => {
    const component = renderer.create(
      <Button></Button>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
