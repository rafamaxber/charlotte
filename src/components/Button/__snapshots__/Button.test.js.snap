// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should match with snapshot 1`] = `
<a
  className="sc-bdVaJa fxfHPe"
  onClick={undefined}
/>
`;
